src/main/java/edu/kis/powp/jobs2d/command/gui/CommandEditorWindow.java

```diff
@@ -0,0 +1,156 @@
+        Container content = this.getContentPane();
+        content.setLayout(new GridBagLayout());
+        content.setBackground(Color.WHITE);
+        this.commandManager = commandManager;
+
+        GridBagConstraints c = new GridBagConstraints();
+        currentCommandField = new JTextArea("");
+        currentCommandField.setEditable(false);
+        c.fill = GridBagConstraints.BOTH;
+        c.gridx = 0;
+        c.gridy = 0;
+        c.weighty = 0.05;
+        c.weightx = 1;
+
+        content.add(currentCommandField, c);
+        updateCurrentCommandField();
+
+        commandPartsList = new JList<>();
+        listModel = new DefaultListModel<>();
+        commandPartsList.setModel(listModel);
+        c.gridy = 1;
+        c.weighty = 0.75;
+        content.add(commandPartsList, c);
+        updateCurrentCommandPartList();
+
+        c.weighty = 0.05;
+
+        c.gridy = 2;
+        moveUp = new JButton("Move command higher ");
+        moveUp.addActionListener((ActionEvent e) -> this.moveUp(commandPartsList.getSelectedValue()));
+        moveUp.setPreferredSize(new Dimension(300, 30));
+        content.add(moveUp, c);
+
+        c.gridy = 3;
+        moveDown = new JButton("Move command lower ");
+        moveDown.setPreferredSize(new Dimension(300, 30));
+        moveDown.addActionListener((ActionEvent e) -> this.moveDown(commandPartsList.getSelectedValue()));
+        content.add(moveDown, c);
+
+        c.gridy = 4;
+        c.fill = GridBagConstraints.HORIZONTAL;
+        Container container = new Container();
+        GridLayout gridLayout = new GridLayout(1, 4);
+        gridLayout.setHgap(25);
+        container.setLayout(gridLayout);
+
+        coordinatesX = new JTextArea("X coordinate ");
+        coordinatesX.setEditable(false);
+        container.add(coordinatesX);
+        coordX = new JTextField("0");
+        container.add(coordX);
+
+        coordinatesY = new JTextArea("Y coordinate ");
+        coordinatesY.setEditable(false);
+        container.add(coordinatesY);
+        coordY = new JTextField("0");
+        container.add(coordY);
+        content.add(container, c);
+
+        c.gridy = 5;
+        c.fill = GridBagConstraints.BOTH;
+        changeCoordinates = new JButton("Change coordinates ");
+        changeCoordinates.setPreferredSize(new Dimension(300, 30));
+        changeCoordinates.addActionListener(
+                (ActionEvent e) -> this.changeCoordinates(Integer.parseInt(coordX.getText()), Integer.parseInt(coordY.getText()),
+                        commandPartsList.getSelectedValue()));
+        content.add(changeCoordinates, c);
+    }
+
+    public void changeCoordinates(int x, int y, DriverCommand command) {
+        int index = commandPartsList.getSelectedIndex();
+        try {
+            listModel.set(index, command.getClass().getDeclaredConstructor(int.class, int.class).newInstance(x, y));
+        } catch (ReflectiveOperationException ignored) { }
```

Comment:
FIX: Using reflection can be replaced with better and safer OO design. In this case, visitor pattern can be used (again) to delegate a creation of a proper driverCommand.